test/NonSilo.Tests/Directory/GrainLocatorResolverTests.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+using NSubstitute;
+using Orleans.GrainDirectory;
+using Orleans.Hosting;
+using Orleans.Runtime;
+using Orleans.Runtime.GrainDirectory;
+using TestExtensions;
+using UnitTests.Grains.Directories;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NonSilo.Tests.Directory
+{
+    [TestCategory("BVT"), TestCategory("Directory")]
+    public class GrainLocatorResolverTests
+    {
+        private readonly IGrainDirectory customDirectory;
+        private readonly IHost host;
+        private readonly GrainLocatorResolver target;
+
+        public GrainLocatorResolverTests(ITestOutputHelper output)
+        {
+            this.customDirectory = Substitute.For<IGrainDirectory>();
+
+            var hostBuilder = new HostBuilder();
+            hostBuilder.UseOrleans((ctx, siloBuilder) =>
+            {
+                siloBuilder
+                    .ConfigureServices(svc => svc.AddSingleton(Substitute.For<DhtGrainLocator>(null, null)))
+                    .ConfigureServices(svc => svc.AddSingletonNamedService(CustomDirectoryGrain.DIRECTORY, (sp, nameof) => this.customDirectory))
+                    .ConfigureLogging(builder => builder.AddProvider(new XunitLoggerProvider(output)))
+                    .UseLocalhostClustering();
+            });
+            this.host = hostBuilder.Build();
```

Comment:
Even though it's not started, we should dispose of this after the tests complete

Reply:
@benjaminpetit we can fix this in a follow-up